dblookupext/miniblocksHandler.go

```diff
@@ -0,0 +1,198 @@
+		err := mh.miniblockHashByTxHashIndexStorer.PutInEpoch(txHash, miniblockHash, epoch)
+		if err != nil {
+			return fmt.Errorf("%w for tx hash %x, miniblock hash %x", err, txHash, miniblockHash)
+		}
+	}
+
+	return nil
+}
+
+func (mh *miniblocksHandler) getExecutedTxHashes(miniblockHeader data.MiniBlockHeaderHandler, mb *block.MiniBlock) [][]byte {
+	txHashes := make([][]byte, 0, len(mb.TxHashes))
+	for index := miniblockHeader.GetIndexOfFirstTxProcessed(); index <= miniblockHeader.GetIndexOfLastTxProcessed(); index++ {
+		txHash := mb.TxHashes[index]
+		txHashes = append(txHashes, txHash)
+	}
+
+	return txHashes
+}
+
+func (mh *miniblocksHandler) saveMiniblockMetadata(
+	header data.HeaderHandler,
+	headerHash []byte,
+	miniblockHeader data.MiniBlockHeaderHandler,
+	miniblockHash []byte,
+	mb *block.MiniBlock,
+) error {
+	miniblockMetaData := &MiniblockMetadataV2{
+		MiniblockHash:          miniblockHash,
+		SourceShardID:          mb.GetSenderShardID(),
+		DestinationShardID:     mb.ReceiverShardID,
+		Type:                   int32(mb.Type),
+		NumTxHashesInMiniblock: int32(len(mb.TxHashes)),
+	}
+
+	var txHasheshenPartial [][]byte
+
+	if miniblockHeader.GetConstructionState() == int32(block.PartialExecuted) {
+		loadedMiniblockMetadata, err := mh.loadExistingMiniblocksMetadata(miniblockHash, header.GetEpoch())
+		if err != nil {
+			return err
+		}
+		if len(loadedMiniblockMetadata.MiniblockHash) != 0 {
+			// rewrite, we found an existing info in storage
+			miniblockMetaData = loadedMiniblockMetadata
+		}
+		txHasheshenPartial = mh.getExecutedTxHashes(miniblockHeader, mb)
+	}
+
+	miniblockOnHeader := &MiniblockMetadataOnBlock{
+		Round:                   header.GetRound(),
+		HeaderNonce:             header.GetNonce(),
+		HeaderHash:              headerHash,
+		Epoch:                   header.GetEpoch(),
+		TxHashesWhenPartial:     txHasheshenPartial,
+		IndexOfFirstTxProcessed: miniblockHeader.GetIndexOfFirstTxProcessed(),
+	}
+
+	miniblockMetaData.MiniblocksInfo = append(miniblockMetaData.MiniblocksInfo, miniblockOnHeader)
+
+	return mh.saveMiniblocksMetadata(miniblockHash, miniblockMetaData, header.GetEpoch())
+}
+
+func (mh *miniblocksHandler) loadExistingMiniblocksMetadata(miniblockHash []byte, epoch uint32) (*MiniblockMetadataV2, error) {
+	multipleMiniblockMetadata := &MiniblockMetadataV2{}
+	buff, err := mh.miniblocksMetadataStorer.GetFromEpoch(miniblockHash, epoch)
+	if err == storage.ErrKeyNotFound {
+		return multipleMiniblockMetadata, nil
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	err = mh.marshaller.Unmarshal(multipleMiniblockMetadata, buff)
+	if err != nil {
+		return mh.tryLegacyUnmarshal(buff)
+	}
+
+	return multipleMiniblockMetadata, err
+}
+
+func (mh *miniblocksHandler) tryLegacyUnmarshal(buff []byte) (*MiniblockMetadataV2, error) {
+	miniblockMetadata := &MiniblockMetadata{}
+	err := mh.marshaller.Unmarshal(miniblockMetadata, buff)
+	if err != nil {
+		return nil, err
+	}
+
+	return &MiniblockMetadataV2{
```

Comment:
yes, the idea is that, when ready for integration, this component will only work with v2 structs. the historyRepository will be converted to work only with v2 structs since they are more general and support partial miniblocks